docs/Tutorial:-How-to-create-a-smart-contract-on-Massa.md

```diff
@@ -0,0 +1,277 @@
+            Storage.set_data("gameState", vec_game_state.join());
+            if (player == "X") {
+                Storage.set_data("currentPlayer", "O");
+            }
+            else {
+                Storage.set_data("currentPlayer", "X");
+            }
+            _checkWin(player)
+        }
+    }
+}
+
+function _checkWin(player: string): void {
+    const winningConditions = [
+        [0, 1, 2],
+        [3, 4, 5],
+        [6, 7, 8],
+        [0, 3, 6],
+        [1, 4, 7],
+        [2, 5, 8],
+        [0, 4, 8],
+        [2, 4, 6]
+    ];
+
+    let game_state = Storage.get_data("gameState");
+    let vec_game_state = game_state.split(",");
+
+    let roundWon = false;
+    for (let i = 0; i <= 7; i++) {
+        const winCondition = winningConditions[i];
+        let a = vec_game_state[winCondition[0]];
+        let b = vec_game_state[winCondition[1]];
+        let c = vec_game_state[winCondition[2]];
+        if (a == "n" || b == "n" || c == "n") {
+            continue;
+        }
+        if (a == b && b == c) {
+            roundWon = true;
+            break
+        }
+    }
+
+    if (roundWon) {
+        Storage.set_data("gameWinner", player);
+    }
+
+    let roundDraw = !vec_game_state.includes("n");
+    if (roundDraw) {
+        Storage.set_data("gameWinner", "draw");
+    }
+}
+
+```
+
+The `play` function is used to update the state of the game when each player plays. As the `initialize` function, it is a public function: anyone can call it and play the next move. Public functions of Massa smart-contracts can only take strings as arguments. To pass several arguments, we thus have to rely on `json-as` and to define the possible arguments using `PlayArgs`.
+
+The `_checkWin` function is used to check whether the game ended or not. Private, as it does not use the `export` prefix, it cannot be called by anyone. It can only be called internally by the smart-contract.
+
+#### main.ts
+
+```ts
+import { Storage, Context, include_base64, call, print, create_sc } from "massa-sc-std";
+import { JSON } from "json-as";
+import { PlayArgs } from "./tic_tac_toe";
+
+function createContract(): string {
+    const bytes = include_base64('./build/tictactoe_play.wasm');
+    const sc_address = create_sc(bytes);
+    return sc_address;
+}
+
+export function main(_args: string): i32 {
+    const sc_address = createContract();
+    print("Created tictactoe smart-contract at:" + sc_address);
+    return 0;
+}
+```
+
+### Compiling your smart-contract
+
+Smart-contract can be compiled using the `massa-sc-scripts` command: TODO.
+
+## Putting your smart-contract on the blockchain
+
+We'll now turn to the process of putting the smart-contract on the Massa blockchain.
+
+### Setup
+
+To put your smart-contract on the Massa blockchain, you need to run a node and use the client. If this is something that you have never done, all the steps are detailed in the [tutorials](https://github.com/massalabs/massa#tutorials-to-join-the-testnet).
```

Comment:
Maybe don't tell that we need to run a node. Just need the client with good configuration.